modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -520,21 +520,16 @@ public void setParameterExampleValue(CodegenParameter p) {
     @Override
     public String getSwaggerType(Property p) {
         String swaggerType = super.getSwaggerType(p);
-        String type;
+
+        // don't apply renaming on types from the typeMapping
         if (typeMapping.containsKey(swaggerType)) {
-            type = typeMapping.get(swaggerType);
-            if (languageSpecificPrimitives.contains(type) || type.indexOf(".") >= 0 ||
-                type.equals("Map") || type.equals("List") ||
-                type.equals("File") || type.equals("Date")) {
-                return type;
-            }
-        } else {
-            type = swaggerType;
+            return typeMapping.get(swaggerType);
```

Comment:
This looks sensible for me, though I fear that it changes the meaning of `typeMapping`, when comparing with most other languages, which still apply `toModelName` after applying the typeMapping, other then when it is in `languageSpecificPrimitives`.  I'm not quite sure what the actual intended purpose of type mapping is.

From the documentation (in README.md):

        --type-mappings <type mappings>
            sets mappings between swagger spec types and generated code types in
            the format of swaggerType=generatedType,swaggerType=generatedType.
            For example: array=List,map=Map,string=String

This is still not totally clear. I guess what we want is "if something is the result of a type mapping, use it as-is and don't transform it further" (so no escaping, sanitizing, prefixing, ...). (This is how it is implemented here.)

@fehguy @wing328 is this a sensible approach for type mapping which we should document in general, and also implement for all other languages this way?